src/virt.c

```diff
@@ -715,9 +758,18 @@ refresh_lists (void)
             xpath_ctx = xmlXPathNewContext (xml_doc);
 
             /* Block devices. */
+	    char * bd_xmlpath=NULL;
+	    if( source == blockdevice_format ){
+		    bd_xmlpath="/domain/devices/disk/source[@dev]";
+	    }else{
+			if( dev == blockdevice_format ){
```

Comment:
The `if` here is confusing, because it suggests that `bd_xmlpath` may stay `NULL`. Instead, I'd suggest to do:

```c
char *bd_xmlpath = (blockdevice_format == source)
    ? "/domain/devices/disk/source[@dev]"
    : "/domain/devices/disk/target[@dev]";
```